utils/utils.py

```diff
@@ -255,3 +262,20 @@ def gpu_state_to_cpu(state_dict):
         if k.startswith('module'):
             new_state_dict[k.replace("module.", "")] = v
     return new_state_dict
+
+
+def on_after_sanity_check(args: Union[argparse.Namespace, DictConfig]):
+    # preprocess paths
+    subs = [args.feature_type]
+    if hasattr(args, 'model_name'):
+        subs.append(args.model_name)
+        # may add `finetuned_on` item
+    real_output_path = args.output_path
+    real_tmp_path = args.tmp_path
+    for p in subs:
+        # some model use `/` e.g. ViT-B/16
+        real_output_path = os.path.join(real_output_path, p.replace("/", "_"))
+        real_tmp_path = os.path.join(real_tmp_path, p.replace("/", "_"))
+    args.output_path = real_output_path
```

Comment:
Ok, I will need to think about this change because I am not a huge fan of patching (changing) the config after it has been loaded. If the bugs occur, it is hard to debug. However, it is just me being cautious.

Reply:
Yes, in fact I also think there may be a more elegant way. I did that because I don't want to change other files too much.

I think it makes sense to treat the path-saving in one place. Because in the future, the newly added models may be trained on different datasets, and then new configuration items may have to be added, and the save path will need to be updated accordingly. 

e.g. model `GoodNet` has three types (`tiny` `base` `large`) and is trained on two datasets (`ImageNet`, `JFT`). One output path might be `output/GoodNet/gnet-base/imagenet`.

Reply:
True. You are right! 

Right now I am not really happy about the structure of this repo. I need to refactor it significantly (configs, models, utils) such that we could avoid so much duplication everywhere for each model which makes it hard to maintain, debug, and add new features.